src/stdlib_utils/parallelism_framework.py

```diff
@@ -223,8 +227,16 @@ def run(
     def _sleep_for_idle_time_during_iteration(
         self, start_timepoint_of_iteration: int
     ) -> None:
-        stop_timepoint_of_iteration = time.perf_counter_ns()
-        iteration_time_ns = stop_timepoint_of_iteration - start_timepoint_of_iteration
+        iteration_time_ns = self.calculate_iteration_time_ns(
+            start_timepoint_of_iteration
+        )
+
+        if len(self._longest_iterations) < 5:
+            self._longest_iterations.append(iteration_time_ns)
+        elif iteration_time_ns > min(self._longest_iterations):
```

Comment:
since you have to refactor to deal with the pylint complain anyway: only calculate `min` once, and also it is faster to reference local variables (since this happens every single loop cycle), so best practice just have once at the top `longest_iterations=self._longest_iterations` then perform all operations on `longest_iterations`